src/lib/consensus/proof_of_stake.ml

```diff
@@ -2876,69 +2691,57 @@ module Hooks = struct
-                    (Mina_base.Frozen_ledger_hash.to_ledger_hash
-                       target_ledger_hash )
-                with
-                | Connected { data = Ok (Ok sparse_ledger); _ } -> (
-                    match
-                      reset_snapshot
-                        ~context:(module Context)
-                        local_state snapshot_id ~sparse_ledger
-                    with
-                    | Ok () ->
-                        (*Don't fail if recording fails*)
-                        don't_wait_for
-                          Trust_system.(
-                            record trust_system logger peer
-                              Actions.(Epoch_ledger_provided, None)) ;
-                        Deferred.Or_error.ok_unit
-                    | Error e ->
-                        [%log faulty_peer_without_punishment]
-                          ~metadata:
-                            [ ("peer", Network_peer.Peer.to_yojson peer)
-                            ; ("error", Error_json.error_to_yojson e)
-                            ]
-                          "Peer $peer failed to serve requested epoch ledger: \
-                           $error" ;
-                        return (Error e) )
-                | Connected { data = Ok (Error err); _ } ->
-                    (* TODO figure out punishments here. *)
-                    [%log faulty_peer_without_punishment]
-                      ~metadata:
-                        [ ("peer", Network_peer.Peer.to_yojson peer)
-                        ; ("error", `String err)
-                        ]
-                      "Peer $peer failed to serve requested epoch ledger: \
-                       $error" ;
-                    return (Or_error.error_string err)
-                | Connected { data = Error err; _ } ->
-                    [%log faulty_peer_without_punishment]
-                      ~metadata:
-                        [ ("peer", Network_peer.Peer.to_yojson peer)
-                        ; ("error", `String (Error.to_string_mach err))
-                        ]
-                      "Peer $peer failed to serve requested epoch ledger: \
-                       $error" ;
-                    return (Error err)
-                | Failed_to_connect err ->
-                    [%log faulty_peer_without_punishment]
-                      ~metadata:
-                        [ ("peer", Network_peer.Peer.to_yojson peer)
-                        ; ("error", Error_json.error_to_yojson err)
-                        ]
-                      "Failed to connect to $peer to retrieve epoch ledger: \
-                       $error" ;
-                    return (Error err) ) )
+            | Error e ->
+                [%log error] "Could not reset snapshot from synced epoch ledger"
+                  ~metadata:[ ("error", Error_json.error_to_yojson e) ] ;
+                return (Error e) )
+        | `Target_changed _ ->
```

Comment:
Because the target hash doesn't change here, this case won't occur, right?
Consequently, during epoch transition, we may end up syncing a couple of times back to back  (since we don't create new goals when hashes change). But I think that should be ok. Starting the sync ledger with the local epoch ledger might reduce the sync time in that case (above comment)

Reply:
I've changed it to an error log. If it doesn't happen, so much the better.

I'm changing the starting ledger to be a snapshot ledger.

Reply:
I've changed it to an error log. If it doesn't happen, so much the better.

I'm changing the starting ledger to be a snapshot ledger.